server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java

```diff
@@ -379,11 +380,14 @@ private ResteasyUriInfo extractUriInfo(final HttpServerRequest req, final String
       if (uri.startsWith(req.scheme() + "://")) {
          uriString = uri;
       } else {
+         String host = req.requestHeaders().get(HttpHeaderNames.HOST);
+         if (host == null || "".equals(host.trim())) {
+            host = "unknown";
```

Comment:
Should we fallback here to something like:
```java
final InetSocketAddress hostAddress = req.hostAddress();
if (hostAddress != null) {
    host = hostAddress.getHostString();
    port = hostAddress.getPort);
} else {
    host = "unknown";
}
```

Reply:
I think the main issue with that it may not be correctly represent what the client-used URI.  The reason I'm changing this is because a customer of ours is complaining about an injected UriInfo having invalid info based on `hostAddress#getHostString` (current code).  Is that worse than `unknown`?  Probably not:)  Anyhow, I think there was some other InetSockAddress option that is more likely to return a correct value, but it comes with a performance hit.  But it has been a bit since I thought about this..

Anyhow, I will think on it, but I'm not opposed to you making a call either.   Our customers probably won't be unhappy with your proposal.

Reply:
My argument for including this is the value here is what is used in any injected `UriInfo` for the request. If a user were to inject this they could do something like:
```java
final URI newUri = uriInfo.getRequestUriBuilder().path("append").buld();
```
That could result in something like "http://unknown/rest/api/append" and give some weird errors. Even worse if unknown actually resolves to something.

Also with HTTP/2 there is not `Host` header.

Reply:
I think that makes sense.  I'm embarrassed I didn't think about 'unknown' resolving to an IP!  I will make this change.  Just a heads up though, this code exists in other server adapters.  There's some duplication amongst these, quite a bit in the tests.  I'm guilty of it:(

Reply:
@jamezp Apologies for going radio silent.  I will get this in within the next few days.

Reply:
Hi @jamezp .  I pushed the change you suggested.